internal/fwserver/server_listresources.go

```diff
@@ -87,17 +121,32 @@ func (s *Server) ListResourceFuncs(ctx context.Context) (map[string]func() list.
 // ListResourceMetadatas returns a slice of ListResourceMetadata for the GetMetadata
 // RPC.
 func (s *Server) ListResourceMetadatas(ctx context.Context) ([]ListResourceMetadata, diag.Diagnostics) {
-	resourceFuncs, diags := s.ListResourceFuncs(ctx)
+	listResourceFuncs, diags := s.ListResourceFuncs(ctx)
 
-	resourceMetadatas := make([]ListResourceMetadata, 0, len(resourceFuncs))
+	listResourceMetadatas := make([]ListResourceMetadata, 0, len(listResourceFuncs))
 
-	for typeName := range resourceFuncs {
-		resourceMetadatas = append(resourceMetadatas, ListResourceMetadata{
+	for typeName := range listResourceFuncs {
+		listResourceMetadatas = append(listResourceMetadatas, ListResourceMetadata{
 			TypeName: typeName,
 		})
 	}
 
-	return resourceMetadatas, diags
+	return listResourceMetadatas, diags
+}
+
+func (s *Server) ListResourceSchema(ctx context.Context, typeName string) (fwschema.Schema, diag.Diagnostics) {
```

Comment:
This method needs a "proper" re-do with locking etc, see `Server.ResourceSchema`.